mlflow/pyfunc/__init__.py

```diff
@@ -1046,17 +1054,65 @@ def _predict_row_batch(predict_fn, args):
         )
         result = predict_fn(pdf)
 
+        if isinstance(result, dict):
+            result = {k: list(v) for k, v in result.items()}
+
         if not isinstance(result, pandas.DataFrame):
             result = pandas.DataFrame(data=result)
 
+        spark_primitive_type_to_np_type = {
+            IntegerType: np.int32,
```

Comment:
Do we want to handle 
`StringType: np.str_` ?

Reply:
Good suggestion! We can support it.